spock-core/src/main/java/org/spockframework/runtime/SpockExecutionContext.java

```diff
@@ -28,6 +34,12 @@
 
   private ErrorInfoCollector errorInfoCollector;
 
+  private final AtomicBoolean hadSuccess = new AtomicBoolean();
+
+  private final List<Throwable> abortions = new CopyOnWriteArrayList<>();
```

Comment:
I'm not really happy that we need to start storing execution results in the context, the `errorInfoCollector` is ugly enough and it was necessary to avoid repeated handling of the same exceptions.

Reply:
Well, I wanted the status of the feature according to the status of the iterations, basically
all iterations skipped => feature skipped
some iterations failing => feature failing
otherwise => feature successful

Reply:
Well, we can store it somewhere else if you prefer.
For example in the `IterationNode`, maybe that is actually even cleaner, yes.
And the `IterationNode`s are stored as children, so I can check those instead in `after`.

Reply:
Besides the still failing tests, I rewrote this part and now store the result where it belongs, inside the iteration node.
Much better code, thx for the push.